homeassistant/components/overkiz/binary_sensor.py

```diff
@@ -145,6 +144,6 @@ class OverkizBinarySensor(OverkizDescriptiveEntity, BinarySensorEntity):
     def is_on(self) -> bool | None:
         """Return the state of the sensor."""
         if state := self.device.states.get(self.entity_description.key):
-            return self.entity_description.value_fn(state.value)
+            return self.entity_description.value_fn(str(state.value))
```

Comment:
```suggestion
            return self.entity_description.value_fn(state)
```

Is state an enum here?  If so we might be able to drop the `str` conversion if `OverkizCommandParam` is the enum its being compared against

Reply:
Unfortunately not. States have a name (str), a type (datatype) and a value (which can be any of the [supported datatypes](https://github.com/iMicknl/python-overkiz-api/blob/326945ae4725dc8575badb4d531e7e0a9905d619/pyoverkiz/enums/general.py#L20-L30)). 

Example of states JSON coming from API:
https://github.com/iMicknl/python-overkiz-api/blob/326945ae4725dc8575badb4d531e7e0a9905d619/tests/fixtures/setup/setup_cozytouch.json#L223-L232

All available datatypes (int enum):
https://github.com/iMicknl/python-overkiz-api/blob/326945ae4725dc8575badb4d531e7e0a9905d619/pyoverkiz/enums/general.py#L20-L30

Reply:
During runtime we could know the type, based on the datatype they provide, however this is something we don't do currently.

Especially because we gather a specific value by name/key, and we know what data type it will provide.

Reply:
Maybe you can adjust the type hint in value_fn?
`value_fn: Callable[[str], bool]` => `value_fn: Callable[[type returned by state.value], bool]`

Reply:
Not sure if I understand what you mean. Can a type hint be dynamic based on another field?

Reply:
You could use this for the definition:
`value_fn: Callable[[None | int | float | str | bool], bool]`

Or maybe add a create and use a type alias:
`value_fn: Callable[[StateValue], bool]`

Reply:
I prefer to have it just typed as a string, and cast it to a string before we pass it to `value_fn` lambda. Otherwise we need to cast/check in every lambda function since we do a str comparison there. Or am I missing something obvious?

Reply:
I think the lazy comparison will work without the need to cast in every lambda, now that overkiz is typed.

Reply:
You are correct! Thanks.